### PR TITLE
add okd images for cnf-network repos

### PR DESCRIPTION
**Pull request by SchSeba**

### CODE REVIEW

**Comment by SchSeba** [comment]
@zeeke @bn222 @wizhaoredhat @fedepaol please review this one :)

**Comment by zeeke** [comment]
[`ci/rehearse/openshift/sriov-network-device-plugin/master/unit-test`](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_release/51044/rehearse-51044-pull-ci-openshift-sriov-network-device-plugin-master-unit-test/1783085789798731776) job has a problem with go vendoring

```
INFO[2024-04-24T11:02:38Z] running gofmt...
Setting GOPATH...
building golint...
go: golang.org/x/lint/golint@latest: cannot query module due to -mod=vendor
```

**Comment by SchSeba** [comment]
Hi @zeeke that is fine I want to enable the lanes I will fix that issues on the repo u/s first after that

File: ci-operator/config/openshift/ib-sriov-cni/openshift-ib-sriov-cni-master__okd-scos.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    ocp_builder_rhel-8-golang-1.21-openshift-4.16:
+      as:
+      - registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.21-openshift-4.16
+    ocp_builder_rhel-9-golang-1.21-openshift-4.16:
+      as:
+      - registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.21-openshift-4.16
+  to: sriov-infiniband-cni
+promotion:
+  to:
+  - name: "4.16"
```
Comment (LorbusChris) [inline ci-operator/config/openshift/ib-sriov-cni/openshift-ib-sriov-cni-master__okd-scos.yaml:29]:
```suggestion
  - name: "scos-4.16"
```
For historical reasons, `origin/4.16` is the OKD/FCOS ImageStream (which we currently don't populate) and `origin/scos-4.16` is the OKD/SCOS this is supposed to get promoted to.

This needs to change here and elsewhere in this PR.

**Review by LorbusChris — CHANGES_REQUESTED** [review]

File: ci-operator/config/openshift/ib-sriov-cni/openshift-ib-sriov-cni-master__okd-scos.yaml (hunk 1 above)
Comment (SchSeba) [reply ci-operator/config/openshift/ib-sriov-cni/openshift-ib-sriov-cni-master__okd-scos.yaml:29]: Hi @LorbusChris I try that but it removes the image from the image_mirror that is all the point of what I need

```
-registry.ci.openshift.org/origin/4.16:metallb quay.io/openshift/origin-metallb:4.16 quay.io/openshift/origin-metallb:4.16.0 quay.io/openshift/origin-metallb:latest
-registry.ci.openshift.org/origin/4.16:metallb-operator quay.io/openshift/origin-metallb-operator:4.16 quay.io/openshift/origin-metallb-operator:4.16.0 quay.io/openshift/origin-metallb-operator:latest
-registry.ci.openshift.org/origin/4.16:metallb-operator-bundle quay.io/openshift/origin-metallb-operator-bundle:4.16 quay.io/openshift/origin-metallb-operator-bundle:4.16.0 quay.io/openshift/origin-metallb-operator-bundle:latest
 registry.ci.openshift.org/origin/4.16:okd-rpms quay.io/openshift/origin-okd-rpms:4.16 quay.io/openshift/origin-okd-rpms:4.16.0 quay.io/openshift/origin-okd-rpms:latest
 registry.ci.openshift.org/origin/4.16:openstack-installer quay.io/openshift/origin-openstack-installer:4.16 quay.io/openshift/origin-openstack-installer:4.16.0 quay.io/openshift/origin-openstack-installer:latest
 registry.ci.openshift.org/origin/4.16:operator-marketplace quay.io/openshift/origin-operator-marketplace:4.16 quay.io/openshift/origin-operator-marketplace:4.16.0 quay.io/openshift/origin-operator-marketplace:latest
 registry.ci.openshift.org/origin/4.16:ovirt-installer quay.io/openshift/origin-ovirt-installer:4.16 quay.io/openshift/origin-ovirt-installer:4.16.0 quay.io/openshift/origin-ovirt-installer:latest
-registry.ci.openshift.org/origin/4.16:sriov-cni quay.io/openshift/origin-sriov-cni:4.16 quay.io/openshift/origin-sriov-cni:4.16.0 quay.io/openshift/origin-sriov-cni:latest
-registry.ci.openshift.org/origin/4.16:sriov-dp-admission-controller quay.io/openshift/origin-sriov-dp-admission-controller:4.16 quay.io/openshift/origin-sriov-dp-admission-controller:4.16.0 quay.io/openshift/origin-sriov-dp-admission-controller:latest
-registry.ci.openshift.org/origin/4.16:sriov-infiniband-cni quay.io/openshift/origin-sriov-infiniband-cni:4.16 quay.io/openshift/origin-sriov-infiniband-cni:4.16.0 quay.io/openshift/origin-sriov-infiniband-cni:latest
-registry.ci.openshift.org/origin/4.16:sriov-network-config-daemon quay.io/openshift/origin-sriov-network-config-daemon:4.16 quay.io/openshift/origin-sriov-network-config-daemon:4.16.0 quay.io/openshift/origin-sriov-network-config-daemon:latest
-registry.ci.openshift.org/origin/4.16:sriov-network-device-plugin quay.io/openshift/origin-sriov-network-device-plugin:4.16 quay.io/openshift/origin-sriov-network-device-plugin:4.16.0 quay.io/openshift/origin-sriov-network-device-plugin:latest
-registry.ci.openshift.org/origin/4.16:sriov-network-metrics-exporter quay.io/openshift/origin-sriov-network-metrics-exporter:4.16 quay.io/openshift/origin-sriov-network-metrics-exporter:4.16.0 quay.io/openshift/origin-sriov-network-metrics-exporter:latest
-registry.ci.openshift.org/origin/4.16:sriov-network-operator quay.io/openshift/origin-sriov-network-operator:4.16 quay.io/openshift/origin-sriov-network-operator:4.16.0 quay.io/openshift/origin-sriov-network-operator:latest
-registry.ci.openshift.org/origin/4.16:sriov-network-webhook quay.io/openshift/origin-sriov-network-webhook:4.16 quay.io/openshift/origin-sriov-network-webhook:4.16.0 quay.io/openshift/origin-sriov-network-webhook:latest

```

**Comment by LorbusChris** [comment]
/hold
pending discussion https://redhat-internal.slack.com/archives/C06QQ74Q2DB/p1714567869555439

File: ci-operator/config/openshift/sriov-dp-admission-controller/openshift-sriov-dp-admission-controller-master__okd-scos.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  to: sriov-dp-admission-controller
+promotion:
+  to:
+  - name: sriov-4.16
+    namespace: origin
+releases:
+  latest:
+    integration:
+      include_built_images: true
+      name: "4.16"
```
Comment (LorbusChris) [inline ci-operator/config/openshift/sriov-dp-admission-controller/openshift-sriov-dp-admission-controller-master__okd-scos.yaml:28]: Let's make sure we always test against the SCOS payload IS
```suggestion
      name: "scos-4.16"
```

**Review by LorbusChris — CHANGES_REQUESTED** [review]

**Comment by SchSeba** [comment]
/pj-rehearse ci/rehearse/openshift/metallb-operator/main/okd-scos-images

File: ci-operator/config/openshift/sriov-network-operator/openshift-sriov-network-operator-master__okd-scos.yaml [hunk 3]
```diff
@@ -3,10 +3,30 @@ base_images:
     name: builder
     namespace: ocp
     tag: rhel-9-golang-1.21-openshift-4.16
+  operator-sdk:
+    name: "4.16"
```
Comment (LorbusChris) [inline ci-operator/config/openshift/sriov-network-operator/openshift-sriov-network-operator-master__okd-scos.yaml:7]:
```suggestion
    name: scos-4.16
```

**Review by LorbusChris — CHANGES_REQUESTED** [review]
Please ensure no references to the origin/4.16 ImageStream are left

**Comment by SchSeba** [comment]
/pj-rehearse pull-ci-openshift-sriov-network-operator-master-okd-scos-ci-bundle-operatorbundle

**Comment by SchSeba** [comment]
cc @jupierce

File: ci-operator/config/openshift/frr/openshift-frr-master__okd-scos.yaml [hunk 4]
```diff
@@ -0,0 +1,44 @@
+base_images:
+  ocp_builder_rhel-9-golang-1.21-openshift-4.16:
+    name: builder
+    namespace: ocp
+    tag: rhel-9-golang-1.21-openshift-4.16
+  origin_scos-4.15_base-stream9:
```
Comment (jupierce) [inline ci-operator/config/openshift/frr/openshift-frr-master__okd-scos.yaml:6]: Can these references be to 4.16? We are doing all of the CentOS migration work in 4.16, so 4.15 might still contain RHEL bits.

**Review by jupierce — CHANGES_REQUESTED** [review]

**Comment by SchSeba** [comment]
/pj-rehearse pull-ci-openshift-metallb-operator-main-okd-scos-images

**Pull request merged**